pkg/cableengine/cableengine.go

```diff
@@ -7,5 +7,5 @@ import (
 type Engine interface {
 	StartEngine() error
 	InstallCable(types.SubmarinerEndpoint) error
-	RemoveCable(string) error
+	RemoveCable(types.SubmarinerEndpoint) error
```

Comment:
While we're changing the interface anyway, I think we should pass a pointer for efficiency to avoid copying. 

Also `SubmarinerEndpoint` merely just wraps an `EndpointSpec` so I think it makes sense to pass that instead. This would make the call site in _tunnel.go_ simpler and cleaner...

Reply:
Thanks for the proposal, however I wrote this patch to comply with the current API as is.

If it indeed makes sense, I'll add more commits to accommodate the requested changes.

My main concern is that the type EndpointSpec is never used directly, but always through a wrapper (i.e. SubmarinerEndpoint). The only exception I see is util.CompareEndpointSpec which references the EndpointSpec type directly.

As for pointer v. copy, again this change was in line with the style of coding in this case. If we want to overhaul the code to be more efficient and use pointers instead of passing copies (the value of which, for performance, IMHO is negligible) then I don't mind doing it in a separate patch.

Reply:
> My main concern is that the type EndpointSpec is never used directly, but always through a wrapper (i.e. SubmarinerEndpoint). The only exception I see is util.CompareEndpointSpec which references the EndpointSpec type directly.

right but I don't see the point for the `SubmarinerEndpoint` wrapper - seems like extra noise and overhead. We can consider removing it in another patch.

> 
> As for pointer v. copy, again this change was in line with the style of coding in this case. If we want to overhaul the code to be more efficient and use pointers instead of passing copies (the value of which, for performance, IMHO is negligible) then I don't mind doing it in a separate patch.

yeah - separate patch is fine - we've changed from pass-by-value to pass-by-pointer in other places in the code base as well.